lib/ecto/adapters/mysql/connection.ex

```diff
@@ -495,7 +495,7 @@ if Code.ensure_loaded?(Mariaex) do
         case elem(sources, pos) do
           {table, schema} ->
             name = String.first(table) <> Integer.to_string(pos)
-            {quote_table(prefix, table), name, schema}
+            {quote_table(prefix || (schema && schema.__schema__(:prefix)), table), name, schema}
```

Comment:
The query has only one prefix that is applied to all tables in SQL query,
this does not work for cross-schema queries, e.g.

```sql
SELECT * FROM "my_prefix"."comments" INNER JOIN "posts" ...
```
In this example `my_prefix` should be applied only to `comments`,
but if it will be set to query it will be set to both `comments` and `posts`.

I've create a test for this case - https://github.com/shhavel/ecto/blob/bdad903e3ad09458099b4a13d53ea4f92393acc8/integration_test/cases/assoc.exs#L162